scipy/stats/_hypotests.py

```diff
@@ -663,3 +663,272 @@ def somersd(x, y=None):
         raise ValueError("x must be either a 1D or 2D array")
     d, p = _somers_d(table)
     return SomersDResult(d, p, table)
+
+
+def _compute_log_combinations(n):
+    """Compute all log combination of C(n, k)"""
+    gammaln_arr = gammaln(np.arange(n + 1) + 1)
+    return gammaln(n + 1) - gammaln_arr - gammaln_arr[::-1]
+
+
+BarnardExactResult = make_dataclass(
+    "BarnardExactResult", [("statistic", float), ("pvalue", float)]
+)
+
+
+def barnard_exact(table, alternative="two-sided", pooled=True, num_it=3):
+    r"""
+    Perform a Barnard exact test on a 2x2 contingency table.
```

Comment:
```suggestion
    r"""Perform a Barnard exact test on a 2x2 contingency table.
```

Reply:
Update in a96a2d3d406fdfbeac238294ac494f618cfa6d9e